Feature/club registration

Chumbita opened the pull request:
Terminé las siguientes funcionalidades: 
1. Login/Register de usuarios (clubes y jugadores) mediante email y validación OTP.
2. Login/Register de jugadores mediante Google OAuth.

Adicional:
3. Se establecieron rutas seguras y guards para impedir el acceso mediante URL. 
4. Refactorización de registerService y hook useRegisterService, el cuál la lógica la trasladé a playerService y clubService y sus correspondientes hooks.

Chumbita committed:
Cambios realizados:
- Conexión con el endopoint de registro de club.
- Corrección de redireccionamientos en el front.
- Refactorización de hooks.

Chumbita committed:
Merge branch 'develop' into feature/club-registration

Chumbita committed:
Mejora de la UI/UX en la página principal (Mobile)

Chumbita committed:
Merge branch 'develop' into feature/club-registration

Chumbita committed:
Se terminó de implementar las siguientes funcionalidades:
- Login/register de clubes y de jugadores mediante email y validación OTP.
- Login/register de jugadores mediante GoogleOAuth.

Otros cambios:
- Mejora de la interfaz de la página Home.
- Refactorización de servicio de Registro (registerService) y hook (useRegisterService).

oliveracarlos commented:
Cuando un jugador hace login con Google, la foto de perfil sigue mostrando la inicial del nombre, en el backend extraemos la URL de la picture que nos da google. Entonces cuando un player se registra con google deberia mostrar la foto de google.

Chumbita committed:
Se realizaron los cambios pedidos:
- Cuando el usuario se registra mediante Google OAuth, en el header debe aparecer la imagen del usuario.

Chumbita commented:
Cambios realizados:
1. Cuando el usuario se registra mediante Google OAuth, en el header debe aparecer la imagen del usuario.

Chumbita committed:
Merge branch 'develop' into feature/club-registration
